### PR TITLE
ENH: Include model revision

### LINKED ISSUE #317: ENH: including model revision into model spec
### Is your feature request related to a problem? Please describe
HF models & scripts can be updated, which may result in model re-downloading or introduce incompatible changes with xinference.

### Describe the solution you'd like
Add a model revison to each pytorch model spec. When caching a model, specify the model revision.

### PR DESCRIPTION
**Pull request by RayJi01**
Resolves #317 

For GGML Model, all Model Revision are set to null for placeholder (later changes may appear and thus may change the revision)
For Pytorch Model, all revision are currently set to the latest commit on HuggingFace.

### CODE REVIEW

File: xinference/model/llm/llm_family.json [hunk 1]
```diff
@@ -257,7 +264,8 @@
           "q8_0"
         ],
         "model_id": "TheBloke/vicuna-33B-GGML",
-        "model_file_name_template": "vicuna-33b.ggmlv3.{quantization}.bin"
+        "model_file_name_template": "vicuna-33b.ggmlv3.{quantization}.bin",
+        "model_revision": null
```
Comment (UranusSeven) [inline xinference/model/llm/llm_family.json:268]: Explicitly specifying null is unnecessary in this case.

Comment (RayJi01) [reply xinference/model/llm/llm_family.json]: every model_revision: null is deleted in the latest version

File: xinference/model/llm/pytorch/core.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def _sanitize_model_config(
     ) -> PytorchModelConfig:
         if pytorch_model_config is None:
             pytorch_model_config = PytorchModelConfig()
-        pytorch_model_config.setdefault("revision", "main")
+        pytorch_model_config["revision"] = self.model_spec.model_revision
```
Comment (UranusSeven) [inline xinference/model/llm/pytorch/core.py:82]: Use `setdefault`.

Comment (RayJi01) [reply xinference/model/llm/pytorch/core.py]: updated in the latest version

**Review by UranusSeven — APPROVED** [review]

**Pull request merged**